Tests/Editor/SettingsAnalysisTests.cs

```diff
@@ -560,5 +560,45 @@ void ResetQualityLevelsValues(List<bool> values)
                 QualitySettings.streamingMipmapsActive = values[i];
             }
         }
+
+        [Test]
+#if !UNITY_2019_3_OR_NEWER
+        [Ignore("This requires the new Shader API")]
+#endif
+        public void SrpAssetSettingsAnalysis_SrpBatching_IsNotReportedOnceFixed()
+        {
+#if UNITY_2019_3_OR_NEWER
+            SrpAssetSettingsAnalyzer.SetSrpBatcherSetting(GraphicsSettings.defaultRenderPipeline);
+            SrpAssetSettingsAnalyzer.SetSrpBatcherSetting(QualitySettings.renderPipeline);
+
+            if (GraphicsSettings.defaultRenderPipeline != null)
+            {
+                TestSrpBatchingSetting(GraphicsSettings.defaultRenderPipeline, -1);
+            }
+
+            if (QualitySettings.renderPipeline != null)
+            {
+                TestSrpBatchingSetting(QualitySettings.renderPipeline, QualitySettings.GetQualityLevel());
+            }
+#endif
+        }
+
+#if UNITY_2019_3_OR_NEWER
+        private void TestSrpBatchingSetting(RenderPipelineAsset renderPipeline, int qualityLevel)
+        {
+            SrpAssetSettingsAnalyzer.SetSrpBatcherSetting(renderPipeline, false);
+            var issues = Analyze(IssueCategory.ProjectSetting,
+                i => i.descriptor.title.Equals("SRP Asset: SRP Batcher"));
+            var srpBatchingIssue = issues.FirstOrDefault();
+            Assert.NotNull(srpBatchingIssue);
+            Assert.IsTrue(issues.Any(i => i.GetCustomPropertyInt32(0) == qualityLevel),
+                $"Render Pipeline should have quality level {qualityLevel}.");
+
+            SrpAssetSettingsAnalyzer.SetSrpBatcherSetting(renderPipeline);
```

Comment:
Similar to the previous comment, it looks as if in the end we set the setting to "true".
So it may be changed after the test if it originally was "false" on the Project.

Reply:
Right, I've fixed the test.